### PR TITLE
typing: more explict "type" typings

### PR DESCRIPTION
**Pull request by onerandomusername**
## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/channel.py [hunk 1]
```diff
@@ -230,9 +230,11 @@ async def _get_channel(self):
         return self
 
     @property
-    def type(self) -> ChannelType:
+    def type(self) -> Literal[ChannelType.text, ChannelType.news]:
         """:class:`ChannelType`: The channel's Discord type."""
```
Comment (shiftinv) [inline disnake/channel.py:234]:
```suggestion
        """:class:`ChannelType`: The channel's Discord type.

        This always returns :attr:`ChannelType.text` or :attr:`ChannelType.news`.
        """
```

File: disnake/channel.py [hunk 2]
```diff
@@ -2952,8 +2971,11 @@ def __repr__(self) -> str:
         return f"<GroupChannel id={self.id} name={self.name!r}>"
 
     @property
-    def type(self) -> ChannelType:
-        """:class:`ChannelType`: The channel's Discord type."""
+    def type(self) -> Literal[ChannelType.group]:
+        """:class:`ChannelType`: The channel's Discord type.
+
+        This always returns :attr:`ChannelType.private`.
```
Comment (shiftinv) [inline disnake/channel.py:2977]:
```suggestion
        This always returns :attr:`ChannelType.group`.
```

File: disnake/channel.py [hunk 3]
```diff
@@ -760,7 +762,9 @@ async def create_thread(
         name: str,
         message: Optional[Snowflake] = None,
         auto_archive_duration: AnyThreadArchiveDuration = None,
-        type: Optional[ChannelType] = None,
+        type: Optional[
+            Literal[ChannelType.public_thread, ChannelType.private_thread, ChannelType.news_thread]
```
Comment (shiftinv) [inline disnake/channel.py:766]: This and the overload above could use `ThreadType` defined in threads.py

File: disnake/threads.py [hunk 4]
```diff
@@ -231,7 +235,7 @@ def _update(self, data):
             pass
 
     @property
-    def type(self) -> ChannelType:
+    def type(self) -> ThreadType:
         """:class:`ChannelType`: The channel's Discord type."""
```
Comment (onerandomusername) [inline disnake/threads.py:239]: i might want to update this docstring too.

File: disnake/channel.py (hunk 3 above)
Comment (onerandomusername) [reply disnake/channel.py]: https://github.com/DisnakeDev/disnake/pull/543/commits/234936ef4fc6725bbf607c02357419c20e99716b

File: disnake/channel.py (hunk 1 above)
Comment (onerandomusername) [reply disnake/channel.py]: https://github.com/DisnakeDev/disnake/pull/543/commits/2d90aec43a03d7eb53fe16112815903425b2e895

File: disnake/threads.py (hunk 4 above)
Comment (onerandomusername) [reply disnake/threads.py]: https://github.com/DisnakeDev/disnake/pull/543/commits/2d90aec43a03d7eb53fe16112815903425b2e895

**Review by shiftinv — COMMENTED** [review]
`RawThreadDeleteEvent.thread_type` should use `ThreadType` as well

**Review by shiftinv — APPROVED** [review]

**Pull request merged**